### DISCUSSION TITLE
sliderfollowcircle enlarging by a lot if the slider is not followed completely

### DISCUSSION THREAD

**Discussion by kaimuu** [ppy/osu#25098] [Q&A]
Not quite sure if this is intentional so I wanted to ask about it since it is currently very distracting and if it is intended I'd love there to be an option to disable it/return to how it is in stable.

This is how the sliderfollowcircle normally looks like
![screen1](https://github.com/ppy/osu/assets/36483368/27233f8b-32f5-436a-bea1-0a7b00d3593e)

and here it is at its largest
![vlc_PNYc1wiH5H](https://github.com/ppy/osu/assets/36483368/4be34890-bf1f-4d25-a7fa-19a67a0bc72b)

The animation if you follow the slider regularly seems to match with stable.

**Comment by peppy** [comment]
Stable also does this if you stop following correctly. The scale matches stable on a check of the code ([stable-ref](https://github.com/peppy/osu-stable-reference/blob/1531237b63392e82c003c712faa028406073aa8f/osu!/GameplayElements/HitObjects/Osu/SliderOsu.cs#L1920-L1923) / [lazer](https://github.com/ppy/osu/blob/7bc8908ca9c026fed1d831eb6e58df7624a8d614/osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyFollowCircle.cs#L53-L54)).

**Reply by kaimuu** [reply]
That's weird, stable also does this yes but it's a lot less of an increase from my experience. Like playing classic mod this the only major thing I notice between both games. It's incredibly jarring. I've never had the sliderfollowcircle expand to such a large degree as it does in lazer. No clue what could be causing this difference.

Also sorry for the very late reply, I'm not super used to github.

**Comment by MaxKruse** [comment]
this is one of my last gripes with lazer as well, and i think i found why this is happening.

My best guess as to what happened here is that, after the leniency-adjustments for LegacyLastTick https://github.com/ppy/osu/issues/22805  , the visuals for the expanding sliderball-radius as described here start playing "too late" - whereas in stable, on judgement of the `LegacyLastTick`, the animation starts playing before the sliderball hits the sliderend, in lazer the animation only plays when the sliderball reaches the sliderend, i hope the screenshots can illustrate this (same skin, imported from stable to lazer):

| Client | Frame Before expanding | Frame On expanding |
| --- | --- | --- |
| Lazer | ![image](https://github.com/ppy/osu/assets/10248041/44a39023-fe0a-434a-926c-61acd06a41c4) | ![image](https://github.com/ppy/osu/assets/10248041/e5359394-ac97-404a-b176-4c099cbc176e) |
| Stable | ![image](https://github.com/ppy/osu/assets/10248041/7d78b0bb-74d8-4b6e-bfa5-2299ee9b1812) | ![image](https://github.com/ppy/osu/assets/10248041/b3bfb6f7-02a7-45c7-9810-2798fad79688) |

As a player this is extremely distracting because despite the sliderjudgement being a 100 (Stable plays the 100-animation, lazer does not), no such indication is given except the sliderball-radius expands. This might be a comfort/expectations thing, but it feels extremely odd as off right now, at least to me.

**Comment by JeromeTT** [comment]
Hi, i'm just a regular player (rank 15k). I mainly play stable. I rarely play lazer.

I've tried to determine my gripe with the new implementation in comparison to what I'm used to in stable. From what I can determine from replays going frame by frame, it appears that:
- In lazer, the follow circle expansion seems to only occur when the slider reaches the end of the slider. In stable, the follow circle expansion seems to occur when the cursor leaves the follow circle, or the player stops holding the slider.
    - This means that in lazer the animation plays _AFTER_ the slider has disappeared, which makes it a lot more noticeable compared to stable.
- In lazer, it seems like the follow circle expansion occurs regardless of slider length. In stable, it appears to me that short sliders don't have follow circle expansion, but instead just disappear. This could just be ScoreV2 difference, i'm not sure
    - When there are lots of short sliders, it's very unpleasant to the eyes to see so many expanding circles (see example below)

I'm unsure what changes to sliders are intended or not in lazer, but personally, I also feel that the slider follow circle is very, very, distracting in lazer's current build.

I've provided an example below which hopefully makes it obvious what the issue is.

Hope this helps. Cheers.
____________________
**Examples are below.** - refer to the slider section after the repeating slider
____________________
 In both clips I treated the slider heads as circles (did not hold the sliders and did not follow the sliders). Both were tested with the "Test" function in the editor.

osu!stable example:

https://github.com/ppy/osu/assets/20854869/61901279-f513-45e8-b5ce-8eb8ccae5f22

osu!lazer example:

https://github.com/ppy/osu/assets/20854869/f2d2a128-a0f4-4a43-984d-d7fe048b9c79
